javascript/tables/src/validationUtils.ts

```diff
@@ -0,0 +1,324 @@
+			valuesCasted.forEach((value: number | Date, index: number) => {
+				if (!validate(value)) resultIndexes.push(index)
+			})
+			return {
+				fail: resultIndexes.length !== 0,
+				indexes: resultIndexes,
+				rule: ErrorCode.Minimum,
+			}
+		} else {
+			return {
+				fail: !(dataType === DataType.Number
+					? (values as number[]).every(validate)
+					: (values as Date[]).every(validate)),
+				indexes: resultIndexes,
+				rule: ErrorCode.Minimum,
+			}
+		}
+	}
+}
+
+export function validateMinimum(
+	minimum: number,
+	dataType: DataType,
+): (value: Date | number) => boolean {
+	return function validateMinimum(value: Date | number) {
+		return dataType == DataType.Date
+			? value.valueOf() >= minimum
+			: value >= minimum
+	}
+}
+
+export function validateMaximumConstraint(
+	maximum: number,
+	dataType: DataType,
+): (values: unknown, includeInstances: boolean) => ValidationTestResult {
+	const validate = validateMaximum(maximum, dataType)
+	return function validateMaximumConstraint(
+		values: unknown,
+		includeInstances: boolean,
+	): ValidationTestResult {
+		const resultIndexes: number[] = []
+		const valuesCasted =
+			dataType === DataType.Number ? (values as number[]) : (values as Date[])
+
+		if (includeInstances) {
+			valuesCasted.forEach((value: number | Date, index: number) => {
+				if (!validate(value)) resultIndexes.push(index)
+			})
+			return {
+				fail: resultIndexes.length !== 0,
+				indexes: resultIndexes,
+				rule: ErrorCode.Maximum,
+			}
+		} else {
+			return {
+				fail: !(dataType === DataType.Number
+					? (values as number[]).every(validate)
+					: (values as Date[]).every(validate)),
+				indexes: resultIndexes,
+				rule: ErrorCode.Maximum,
+			}
+		}
+	}
+}
+
+export function validateMaximum(
+	maximum: number,
+	dataType: DataType,
+): (value: Date | number) => boolean {
+	return function validateMaximum(value: Date | number) {
+		return dataType == DataType.Date
+			? value.valueOf() <= maximum
+			: value <= maximum
+	}
+}
+
+export function validatePatternConstraint(
+	pattern: string,
+): (values: unknown, includeInstances: boolean) => ValidationTestResult {
+	const validate = validatePattern(pattern)
+	return function validatePatternConstraint(
+		values: unknown,
+		includeInstances: boolean,
+	): ValidationTestResult {
+		const resultIndexes: number[] = []
+		const valuesCasted = values as string[]
+
+		if (includeInstances) {
+			valuesCasted.forEach((value: string, index: number) => {
+				if (!validate(value)) resultIndexes.push(index)
+			})
+			return {
+				fail: resultIndexes.length !== 0,
+				indexes: resultIndexes,
+				rule: ErrorCode.Pattern,
+			}
+		} else {
+			return {
+				fail: !valuesCasted.every(validate),
+				indexes: resultIndexes,
+				rule: ErrorCode.Pattern,
+			}
+		}
+	}
+}
+
+export function validatePattern(pattern: string): (value: string) => boolean {
+	return function validatePattern(value: string) {
+		const re = new RegExp(pattern)
```

Comment:
Try moving the RegExp outside of the closure so it only needs to be created once. However, make sure this works ok because some RegExp methods are stateful. Whether you can do this depends on the usage (whether the RegExp is global).